homeassistant/components/dwd_weather_warnings/__init__.py

```diff
@@ -3,22 +3,57 @@
 from __future__ import annotations
 
 from dwdwfsapi import DwdWeatherWarningsAPI
+import voluptuous as vol
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
 
-from .const import CONF_REGION_IDENTIFIER, DOMAIN, PLATFORMS
+from .const import (
+    CONF_REGION_DEVICE_TRACKER,
+    CONF_REGION_IDENTIFIER,
+    DOMAIN,
+    LOGGER,
+    PLATFORMS,
+)
 from .coordinator import DwdWeatherWarningsCoordinator
+from .exceptions import EntityNotFoundError
+from .util import get_position_data
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up a config entry."""
-    region_identifier: str = entry.data[CONF_REGION_IDENTIFIER]
+    region_identifier: str = entry.data.get(CONF_REGION_IDENTIFIER, None)
+    device_tracker: str = entry.data.get(CONF_REGION_DEVICE_TRACKER, None)
 
-    # Initialize the API and coordinator.
-    api = await hass.async_add_executor_job(DwdWeatherWarningsAPI, region_identifier)
-    coordinator = DwdWeatherWarningsCoordinator(hass, api)
+    # Initialize the API and coordinator based on the specified data.
+    if region_identifier is not None:
+        api = await hass.async_add_executor_job(
+            DwdWeatherWarningsAPI, region_identifier
+        )
+    elif device_tracker is not None:
+        registry = er.async_get(hass)
+
+        try:
+            device_tracker = er.async_validate_entity_id(registry, device_tracker)
+        except vol.Invalid:
+            # The entity/UUID is invalid or not associated with an entity registry item.
+            LOGGER.error(
+                "Failed to setup dwd_weather_warnings for unknown entity %s",
+                device_tracker,
+            )
+            return False
+
+        try:
+            position = get_position_data(hass, device_tracker)
+        except (EntityNotFoundError, AttributeError) as err:
+            # The provided device_tracker is not available or missing required attributes.
+            LOGGER.error(f"Failed to setup dwd_weather_warnings: {repr(err)}")
+            return False
 
+        api = await hass.async_add_executor_job(DwdWeatherWarningsAPI, position)
```

Comment:
All of this seems like duplicated logic which the coordinator also needs to handle, can you try to make it happen as a result of calling `DwdWeatherWarningsCoordinator.async_config_entry_first_refresh` instead?

Reply:
I'll see what I can do, not familiar with that method though

Reply:
Took a few tries but I got it (see last 3 commits)